src/integrations/pathManager.ts

```diff
@@ -0,0 +1,99 @@
+import fs from 'fs';
+import os from 'os';
+import path from 'path';
+import paths from '@/utils/paths';
+import manageLinesInFile from '@/integrations/manageLinesInFile';
+
+let pathManager: ManualPathManager;
+
+// PathManager is the interface that anything that manages the
+// PATH variable must implement.
+interface PathManager {
+  // Makes real any changes to the system. Should be idempotent.
+  enforce(): Promise<void>
+  // Removes any changes that the PathManager may have made.
+  // Should be idempotent.
+  remove(): Promise<void>
+}
+
+// ManualPathManager is for when the user has chosen to manage
+// their PATH themselves. It does nothing.
+class ManualPathManager implements PathManager {
+  async enforce(): Promise<void> {}
+  async remove(): Promise<void> {}
+}
+
+// RcFilePathManager is for when the user wants Rancher Desktop to
+// make changes to their PATH by putting lines that change it in their
+// shell .rc files.
+export class RcFilePathManager implements PathManager {
+  constructor() {
+    const platform = os.platform();
+
+    if (platform !== 'linux' && platform !== 'darwin') {
+      throw new Error(`Platform "${ platform }" is not supported by RcFilePathManager`);
+    }
+  }
+
+  async enforce(): Promise<void> {
+    await this.managePosix(true);
+    await this.manageCsh(true);
+    await this.manageFish(true);
+  }
+
+  async remove(): Promise<void> {
+    await this.managePosix(false);
+    await this.manageCsh(false);
+    await this.manageFish(false);
+  }
+
+  protected async managePosix(desiredPresent: boolean): Promise<void> {
+    const pathLine = `export PATH="$PATH:${ paths.integration }"`;
```

Comment:
It looks like we are appending the integrations at the end of the `PATH`.

I was under the assumption that we would be prepending to the front, for the benefit of the inexperienced user who might have an older version of the utility already somewhere else in their `PATH`.

We offer the "do-it-yourself" option for users who want to precisely control shadowing of binaries on the `PATH`.

Reply:
That's fair. I've now put the integrations directory at at the head of the `PATH`.